packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -190,8 +181,7 @@ export const updateItemInCart = (payload = {}) => {
                     // Add the updated item to that cart.
                     await dispatch(
                         addItemToCart({
-                            ...payload,
-                            fetchCartId
+                            ...payload
```

Comment:
This concerns me. We have a case in `updateItemInCart` where we may call `addItemToCart` in the case of a newly created cart. The payload for `addItemToCart` is actually changed in this PR so this code no longer actually works.

We would have to somehow do all the logic in the [handleAddToCart](https://github.com/magento/pwa-studio/pull/1987/files#diff-27ab3c434634ee4e2b37a99c7a9772f8R191) callback right here _or_ do it wherever we invoke the `updateItemInCart` action.

Maybe we should do the variant matching and typename conditionals in the `addItemToCart` action. Then, this should function still...

Looking for thoughts.

Reply:
Moving the content of the callback into the action should work but it'll require that any user of `updateItemInCart` pass the mutations for adding items as well. 😭

Reply:
Ended up simplifying this by reverting most of the logic. I just needed to deconstruct the original payloads better in `addItem`. Now `updateItemInCart` also has the `addItem` mutations in the payload so if we get here we are fine.